### PR TITLE
Expand add-region deep link and custom region form with sidecar and Umami analytics support

### PR DESCRIPTION
**Pull request by aaronbrethorst**
## Summary

Expands the `onebusaway://add-region` custom URL scheme and the Add/Edit Custom Region form to optionally configure two region capabilities that previously could not be set on custom regions:

- **Obaco sidecar server URL** (`Region.sidecarBaseURL`) — powers features like alarms and surveys
- **Umami analytics** (`Region.umamiAnalytics`) — analytics server URL + website ID

### Deep link format

```
onebusaway://add-region?name=REGION_NAME&oba-url=ENCODED_OBA_URL
    [&otp-url=ENCODED_OTP_URL]
    [&sidecar-url=ENCODED_SIDECAR_URL]
    [&umami-url=ENCODED_UMAMI_URL&umami-id=UMAMI_WEBSITE_ID]
```

All URL values must be percent-encoded (an unencoded `&` inside a nested URL truncates it — locked in by raw-string decode tests). Umami is both-or-nothing: the config is built only when both `umami-url` and `umami-id` are present and valid; a partial pair is silently ignored and the region still saves. The rule lives in exactly one place — a new failable `UmamiAnalyticsConfig.init?(url:id:)` — called by the deep link path and the form alike.

### Form changes

Two new sections (Sidecar Server, Analytics) in the custom region form, with well-formedness-only validation (Save still gates only on the base URL; the base URL keeps its live server validation). URL normalization was extracted into a general `normalizeURL` helper with `normalizeBaseURL` wrapping it for the `/api/where`-stripping behavior.

### Bug fixes along the way

- Editing a custom region no longer drops values the form doesn't edit: `openTripPlannerURL` (pre-existing bug — a region added via deep link with an OTP URL lost it on any edit) and the new sidecar/umami fields are carried forward via `setInitialValues` repopulation and explicit init arguments.
- `normalizeBaseURL` now re-validates after stripping `/api/where`, so pathological input like `api/where` is rejected instead of yielding a host-less `https:` URL that enabled Save.

## Test plan

- Full OBAKitTests suite: **1336/1336 passing** on the final HEAD; SwiftLint 0 violations.
- New coverage: `UmamiAnalyticsConfig` init edge cases; `Region` Codable round-trip with sidecar+umami (plus backward-compat: old persisted regions decode with nils); URLSchemeRouter decode for all params, partial pairs, invalid URLs, and raw-string percent-encoding contracts; `normalizeURL`/`normalizeBaseURL` regression cases.

## Follow-ups (not in this PR)

- Form-level tests for `setInitialValues` repopulation and `doSave` carry-forward would require extracting the form's state into an inspectable view model; the behavior is code-reviewed but untested at the form level.
- Run `scripts/extract_strings` to get the 5 new `custom_region_builder_controller.*` keys in front of the 13 locales' translators (runtime falls back to the `OBALoc` default values meanwhile).
- Pre-existing deep-link oddity worth a tracking issue: the added region's map coordinate is derived from the *current* region's agency list (`Application.swift`), not the new server's.
- Deep links accept path-only values (e.g. `sidecar-url=/foo`) per long-standing `oba-url`/`otp-url` semantics; worth revisiting if Obaco request construction chokes on host-less URLs.

## Summary by CodeRabbit

* **New Features**
  * Added Sidecar Server configuration to the custom region form.
  * Added Umami analytics URL and website ID configuration.
  * Extended `add-region` deep links to support Sidecar and Umami parameters.
  * Updated custom region editing and saving to persist these settings.
* **Bug Fixes**
  * Improved URL normalization/validation (including stricter handling of invalid inputs and `/api/where` base URLs).
  * Umami analytics now follows an “all-or-nothing” rule; partial/invalid inputs are ignored.
* **Documentation**
  * Updated deep-link documentation with optional parameter behavior and percent-encoding guidance.
  * Refreshed Swift/toolchain guidance in the main documentation.

### CODE REVIEW

File: OBAKitCore/Models/Region.swift [hunk 1]
```diff
@@ -541,6 +543,21 @@ public struct UmamiAnalyticsConfig: Codable, Equatable, Hashable {
...
+        self.id = id
+    }
+
+    /// The single source of truth for the both-or-nothing rule: a config exists
+    /// only when both a URL and a non-blank website ID are present. Partial
+    /// pairs collapse to `nil`, which means "analytics disabled."
+    public init?(url: URL?, id: String?) {
+        guard let url, let id = id?.strip(), !id.isEmpty else {
+            return nil
+        }
```
[the remark this answers is not in the document]
Comment (aaronbrethorst) [reply OBAKitCore/Models/Region.swift:558]: Both constructs are already established in this codebase, so this doesn't block compilation: `url.host()` exists on `main` in this same function (pre-refactor `RegionCustomForm.swift:61`), shorthand optional binding is used repo-wide (e.g. `Apps/Shared/Analytics/UmamiAnalytics.swift:27`), and the targets build with Swift 5/6 language modes on a current toolchain with an iOS 18.0 deployment target (`URL.host()` needs iOS 16+). The "Swift 5.3+" guideline in CLAUDE.md was a stale doc line from an earlier era of the project — updated in d0c494bb so it stops suggesting otherwise. No code change needed; the CI build check compiles this exact code.
